rainier-compute/src/main/scala/com/stripe/rainier/compute/Cholesky.scala

```diff
@@ -0,0 +1,86 @@
+package com.stripe.rainier.compute
+
+/*
+This represents an NxN matrix A
+by the  N(N+1)/2 packed lower-triangular elements
+of its Cholesky decomposition L
+(where A = LL*)
+*/
+case class Cholesky(packed: Vector[Real]) {
+    val rank: Int = Cholesky.rank(packed.size)
+    
+    //log(det(A))
+    def logDeterminant: Real = Real.sum(diagonals.map(_.log)) * 2
+
+    //solve for x where Ax=y
+    def inverseMultiply(y: Vector[Real]): Vector[Real] = {
+        //L*Lx = y, let b=Lx 
+        //solve L*b = y
+        val b = Cholesky.upperTriangularSolve(packed, y)
+        //solve Lx = b
+        Cholesky.lowerTriangularSolve(packed, b)
+    }
+
+    //diagonal elements of L
+    private def diagonals: Vector[Real] = 
+        0.until(rank).toVector.map{i =>
+            packed(Cholesky.triangleNumber(i))
+        }
+}
+
+object Cholesky {
+    def lowerTriangularMultiply(packed: Array[Double], vector: Array[Double]): Array[Double] = {
+        val n = vector.size
+        val result = new Array[Double](n)
+        var i = 0
+        var base = 0
+        while(i < n) {
+            var j = 0
+            var acc = 0d
+            while(j <= i) {
+                acc += vector(j) * packed(base + j)
+                j += 1
+            }
+            result(i) = acc
+            i += 1
+            base += i                
+        }
+        result
+    }
+    
+    def triangleNumber(k: Int) = (k * (k+1)) / 2
```

Comment:
weird spacing: why doe `*` get spaces but not `+`?

Also I'd add a return type. Lastly I'd add a comment: this is the number of non-zero items in a triangluar square matrix with k columns or rows.